src/libqof/qof/gnc-datetime.cpp

```diff
@@ -65,15 +65,15 @@ static constexpr auto ticks_per_second = INT64_C(1000000);
 static constexpr auto ticks_per_second = INT64_C(1000000000);
 #endif
 
-/* Map of date formats understood by gnucash and corresponding regex
+/* Vector of date formats understood by gnucash and corresponding regex
  * to parse each from an external source
  * Note: while the format names are using a "-" as separator, the
  * regexes will accept any of "-/.' " and will also work for dates
  * without separators.
  */
-static std::map<const std::string, const char *> gnc_date_fmt_map = {
-    {
-        std::string(N_("y-m-d")),
+const std::vector<GncDateFormat> GncDate::formats = {
+    GncDateFormat {
```

Comment:
It's a better optimization indeed. That will make it use the copy constructor instead of copy assignment.

Other than that it doesn't seem to make much of a difference. The abort remains.